test/autoscaling.test.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const assume = require('assume');
+const async = require('async');
+const nock = require('nock');
+
+const common = require('./common');
+const beforeTesting = common.beforeTesting;
+const resourceName = 'test-name';
+
+describe('lib.autoscaling', () => {
+  describe('.autoscaling', () => {
+    const path = `/apis/autoscaling/v1/namespaces/${ common.currentName }/horizontalpodautoscalers`;
+    const resourcePath = `${ path }/${ resourceName }`;
+    const horizontalAutoscaleObj = {
+      apiVersion: 'autoscaling/v1',
+      kind: 'HorizontalPodAutoscaler',
+      metadata: {
+        name: 'test-name',
+        namespace: 'test-namespace'
+      },
+      spec: {
+        targetCPUUtilizationPercentage: 65,
+        maxReplicas: 10,
+        minReplicas: 3,
+        scaleTargetRef: {
+          apiVersion: 'autoscaling/v1beta1',
+          kind: 'Deployment',
+          name: 'example-name',
+          subresource: 'scale'
+        }
+      },
+      status: {
+        currentReplicas: 0,
+        desiredReplicas: 0
+      }
+    };
+
+    beforeTesting('int', common.changeName);
+    beforeTesting('unit', () => {
+      nock(common.autoscaling.url)
+        .post(path)
+        .reply(201, horizontalAutoscaleObj)
+        .get(resourcePath)
+        .reply(200, horizontalAutoscaleObj)
+        .delete(resourcePath)
+        .reply(200, horizontalAutoscaleObj);
+    });
+
+    it('POSTs, GETs, and DELETEs', done => {
+      async.series([
+        next => {
+          common.autoscaling.ns.hpa.post({ body: horizontalAutoscaleObj }, next);
+        },
+        next => common.autoscaling.ns.hpa.get(resourceName, next),
+        next => common.autoscaling.ns.hpa.delete(resourceName, next)
+      ], (err, results) => {
+        assume(err).is.falsy();
+        const ds = results[0];
+        assume(ds.metadata.name).is.equal(resourceName);
+        done();
+      });
+    });
+
+    describe('lists', () => {
+      beforeTesting('int', common.changeName);
+      beforeTesting('unit', () => {
+        nock(common.autoscaling.url)
+        .get(path)
+        .reply(200, horizontalAutoscaleObj);
+      });
+
+      it('returns HorizontalPodAutoscalerList', done => {
+        common.autoscaling.ns.hpa.get((err, result) => {
+          assume(err).is.falsy();
+          assume(result.kind).is.equal('HorizontalPodAutoscaler');
```

Comment:
This should be a `HorizontalPodAutoscalerList`, yeah?

Reply:
Yes, added an update the test data to be hpa (Horizontal Pod Autoscaler), rather than ds (daemonsets)

Reply:
Sorry, I meant that `common.autoscaling.ns.hpa.get((err, result) => {` is going to call the callback with a result of `kind` `HorizontalPodAutoscalerList`, not `HorizontalPodAutoscaler`.  I caught this by running the integration tests locally.  I think you need to update the `nock`ing to return a `HorizontalPodAutoscalerList`; and then change the assume statement to be `assume(result.kind).is.equal('HorizontalPodAutoscalerList');`